CMakeLists.txt

```diff
@@ -757,8 +757,9 @@ target_link_libraries (seastar
     lksctp-tools::lksctp-tools
     rt::rt
     yaml-cpp::yaml-cpp
-    Valgrind::valgrind
-    Threads::Threads)
+    Threads::Threads
+  INTERFACE
+    Valgrind::valgrind)
```

Comment:
i still don't understand why we need to have `Valgrind::valgrind` listed as an INTERFACE library. does it address any issue? or what make it an improvement?

Reply:
The issue that is resolved that currently there is transitive unneeded dependency.
Then you use seastar with you code cmake tries to link you code with valgrind. 
The valgrind manual claims that is not necessary. You only need headers from it when you build seastar.
If you need valgrind to work you should explicitly preload it.

Reply:
@tchaikov 
For build on my machine Ubuntu 22.04 build fails as it tries to link with "Valgrind::valgrind" library but it fails.

Reply:
> i am confused. if `Valgrind::valgrind` is a PRIVATE library linked by Seastar, then the Seastar application linked against Seastar does not link against `Valgrind::valgrind`. and an "INTERFACE" library would makes it a transitive library.

Your point makes sense. Still I'm also confused how to remove valgrind from linker dependencies

Reply:
> @tchaikov
> For build on my machine Ubuntu 22.04 build fails as it tries to link with "Valgrind::valgrind" library but it fails.

when it fails? which one fails to link?  Seastar tests/demos/apps fail to link or your application fails to link? could you share the error messages when you have FTBFS?

Reply:
@tchaikov
I'm considering adding **target_include_directories** with PRIVATE Valgrind::valgrind secrion. I will notify you about results.

Reply:
@tchaikov
New solution with [target_include_directories](https://cmake.org/cmake/help/latest/command/target_include_directories.html#command:target_include_directories) seems to be working. I cannot find any clues of valgrind library at _lib/cmake/Seastar/SeastarTargets.cmake_ file in installation directory. Could you please verify solution?

Reply:
this does not make sense. per https://cmake.org/cmake/help/latest/command/target_include_directories.html , `target_include_directories()` accepts *paths* instead targets as its "item" argument.

without the error message and a root cause analysis, i am not able to verify the solution.